lib/eyeshade/last_settlement_balance.rb

```diff
@@ -0,0 +1,38 @@
+module Eyeshade
+  class LastSettlementBalance < BaseBalance
+    attr_reader :date, :amount_settlement_currency, :timestamp, :settlement_currency
+
+    def initialize(rates, default_currency, transactions)
+      super(rates, default_currency)
+
+      last_settlement = calc_last_settlement(transactions)
+      @amount_bat = last_settlement["amount_bat"]
+      @timestamp = last_settlement["timestamp"]
+      @settlement_currency = last_settlement["currency"]
+      @amount_settlement_currency = convert(@amount_bat, @settlement_currency)
+    end
+
+    private
+
+    def calc_last_settlement(transactions)
+      return {} if transactions == []
+       # Find most recent settlement transaction
+      last_settlement_date = transactions.select { |transaction|
```

Comment:
Let's change this to be `do |transaction|` instead of a `{ |transaction|`

Reply:
I used this syntax so I can chain operations to the result of the block, e.g. the `.last["created_at"].to_date`:

```ruby
last_settlement_date = transactions.select { |transaction|
  transaction["settlement_amount"].present?
}.last["created_at"].to_date
```

When I wrote this I didn't think you could achieve the same thing with `do ... end`, but apparently you can.  I think it looks kind of funky though.

```ruby
last_settlement_date = transactions.select  do |transaction|
  transaction["settlement_amount"].present?
end.last["created_at"].to_date
```

What do you think?

Reply:
I think we should just do something like this

```ruby
last_settlement_date = transactions
	.select { |t| t["settlement_amount"].present? }
	.last["created_at"].to_date
```